src/contracts/core/DelegationManager.sol

```diff
@@ -442,6 +442,60 @@ contract DelegationManager is Initializable, OwnableUpgradeable, Pausable, Deleg
         }
     }
 
+    /**
+     * @notice Called by AVSs to register an operator with the AVS.
+     * @param operator The address of the operator to register.
+     * @param signatureWithSaltAndExpiry The signature, salt, and expiry of the operator's signature.
+     */
+    function registerOperatorWithAVS(address operator, ISignatureUtils.SignatureWithSaltAndExpiry memory signatureWithSaltAndExpiry) external {
+        require(signatureWithSaltAndExpiry.expiry > block.timestamp, "DelegationManager.registerOperatorWithAVS: signature expired");
+        require(operatorSaltIsSpent[operator][signatureWithSaltAndExpiry.salt] == false, "DelegationManager.registerOperatorWithAVS: salt already used");
+        require(!registeredWithAVS[msg.sender][operator], "DelegationManager.registerOperatorWithAVS: operator already registered");
+
+        // Calculate the digest hash
+        bytes32 operatorRegistrationDigestHash = calculateOperatorRegistrationDigestHash({
+            operator: operator, 
+            avs: msg.sender, 
+            salt: signatureWithSaltAndExpiry.salt, 
+            expiry: signatureWithSaltAndExpiry.expiry
+        });
+
+        // Check that the signature is valid
+        EIP1271SignatureUtils.checkSignature_EIP1271(
+            operator,
+            operatorRegistrationDigestHash,
+            signatureWithSaltAndExpiry.signature
+        );
+
+        // mark the salt as spent
+        operatorSaltIsSpent[operator][signatureWithSaltAndExpiry.salt] = true;
+        // register operator
+        registeredWithAVS[msg.sender][operator] = true;
+
+        emit OperatorRegistrationStatusUpdatedForAVS(operator, msg.sender, OperatorRegistrationStatus.REGISTERED);
+    }
+
+    /**
+     * @notice Called by AVSs to deregister an operator with the AVS.
+     * @param operator The address of the operator to deregister.
+     */
+    function deregisterOperatorFromAVS(address operator) external {
+        require(registeredWithAVS[msg.sender][operator], "DelegationManager.deregisterOperatorFromAVS: operator not registered");
+
+        // deregister operator
+        registeredWithAVS[msg.sender][operator] = false;
+
+        emit OperatorRegistrationStatusUpdatedForAVS(operator, msg.sender, OperatorRegistrationStatus.DEREGISTERED);
+    }
+
+    /**
+     * @notice Called by an AVS to emit an `OperatorMetadataURIUpdated` event indicating the information has updated.
+     * @param metadataURI The URI for metadata associated with an AVS
+     */
+    function updateAVSMetadataURI(string calldata metadataURI) external {
```

Comment:
who calls it? as in how does AVS update their metadata, via a frontend or something else?

Reply:
This is called by an AVS' registry coordinator contract:

https://github.com/Layr-Labs/eigenlayer-middleware/blob/5e695c1d6c898c394d8810c111777881ee0fa61d/src/BLSRegistryCoordinatorWithIndices.sol#L376-L383